Mathlib/Data/List/Nodup.lean

```diff
@@ -388,6 +389,22 @@ theorem Nodup.take_eq_filter_mem [DecidableEq α] :
     intro x hx
     have : x ≠ b := fun h => (nodup_cons.1 hl).1 (h ▸ hx)
     simp (config := {contextual := true}) [List.mem_filter, this, hx]
+
+theorem setOf_eq_singleton_iff_of_nodup (H : Nodup l) {a : α} [DecidableEq α]
```

Comment:
I think a better theorem here is that converting a list to a set in this way is injecting iff l.Nodup, from which this follows trivially

Reply:
I used permutations instead of `InjOn`, but it should still have the right idea.